staging/src/k8s.io/component-base/logs/json/json.go

```diff
@@ -35,32 +35,23 @@ var (
 	timeNow = time.Now
 )
 
-type noopInfoLogger struct{}
-
-// Enabled func in noopInfoLogger always return false
-func (l *noopInfoLogger) Enabled() bool {
-	return false
-}
-
-func (l *noopInfoLogger) Info(_ string, _ ...interface{}) {}
-
-var disabledInfoLogger = &noopInfoLogger{}
-
-type infoLogger struct {
-	lvl zapcore.Level
+// zapLogger is a logr.Logger that uses Zap to record log.
+type zapLogger struct {
```

Comment:
I am curious why we are embedding this rather than using https://github.com/go-logr/zapr ?

Reply:
There are some small differences configuring zap, e.g. EncoderConfig. As codebase was small we decided that it will be easier and faster to first develop it in k/k and propose changes to zapr later.